### DISCUSSION TITLE
ja/integrations/sony-imx500/

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#18061] [Docs]
# ja/integrations/sony-imx500/

Ultralytics YOLOv8  モデルをソニーの IMX500 フォーマットにエクスポートして、モデルを最適化し、効率的に配備する方法を学びます。

https://docs.ultralytics.com/ja/integrations/sony-imx500/

**Comment by sato-kenryo** [comment]
Can I train YOLOv8n and export it for imx500?
I tried the below code to train and load the model on google colab.
```
model = YOLO("yolov8n.pt")
result = model.train(data="coco8.yaml", epochs=100, imgsz=640)
trained_model = YOLO("/content/runs/detect/train/weights/best.pt")
```

Then run the below code.
```
trained_model.export(format="imx")  # Exports with PTQ quantization by default
```
But I got this error.
```
ModuleNotFoundError: No module named 'keras._tf_keras'
```
Does anyone know solution?
Thank you.

**Reply by glenn-jocher** [reply]
@sato-kenryo the error `ModuleNotFoundError: No module named 'keras._tf_keras'` likely arises due to an incompatibility or missing TensorFlow/Keras installation required for export to the IMX500 format. Ensure you have TensorFlow installed (`pip install tensorflow`) and compatible with your Colab environment. Additionally, refer to the [IMX500 export guide](https://docs.ultralytics.com/integrations/sony-imx500/) for detailed setup instructions. Let us know if the issue persists!

**Reply by sato-kenryo** [reply]
Thank you for advice!
As you said, I checked TensorFlow installation. And I had already had TensorFlow==2.17.0.
But when I run the code below, TensorFlow==2.12.0 was reinstalled and still got the same error.
```
model.export(format="imx")
```
Instead when I tried CLI on gogle colab, it worked!
```
!yolo export model="/content/runs/detect/train/weights/best.pt" format=imx
```
Could it happen?

![image](https://github.com/user-attachments/assets/0936d844-843a-4e16-933b-277b94f0f268)
![image](https://github.com/user-attachments/assets/a6060855-1fb4-4678-86a8-0c11ca4597c1)
![image](https://github.com/user-attachments/assets/5ef1100a-a50b-458d-aedc-39cd3b4cc0be)

**Reply by glenn-jocher** [reply]
It seems that the issue you're encountering with the Python API is likely tied to specific dependencies or the environment setup in Colab. The resolution using the CLI supports that the environment with the CLI command is better configured for the IMX500 export process. I recommend continuing with the CLI workflow for export, as it appears to work seamlessly. For further troubleshooting with the Python API, double-check TensorFlow/Keras versions and ensure all dependencies required for IMX500 export are correctly installed as outlined in the [IMX500 Export Guide](https://docs.ultralytics.com/integrations/sony-imx500/). Let us know if you encounter additional issues!

**Reply by Y-T-G** [reply]
@sato-kenryo You can try exporting on Google Colab.

**Comment by UltralyticsAssistant** [comment]
👋 Hello, thank you for sharing this discussion and your interest in Ultralytics 🚀! It's great to see enthusiasm for exploring integrations like exporting YOLOv8 models to the Sony IMX500 format.

We recommend checking out the [Ultralytics Docs](https://docs.ultralytics.com) for detailed guidance, examples, and tutorials. Specifically, the [Integrations](https://docs.ultralytics.com/integrations/) section may have what you're looking for. You can also find valuable information about exporting and optimizing models.

If you encounter any issues along the way or if this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) so we can better assist you.

If you are exploring a custom training ❓ Question, please include as much detail as possible, such as dataset specifics, logs, and any related observations. Be sure to review our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) to ensure you're achieving optimal performance.

For additional community support or real-time interactions, check out the following resources:
- [Discord](https://discord.com/invite/ultralytics) 🎧
- [Discourse](https://community.ultralytics.com) for in-depth discussions
- [Subreddit](https://reddit.com/r/Ultralytics) to collaborate and share ideas

## Upgrade

Make sure you are using the latest version of the `ultralytics` package. Upgrade your installation in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

To efficiently run or deploy YOLO, consider one of these verified environments with all necessary dependencies pre-installed:
- **Notebooks** with GPU support: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM: Review the [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI: See the [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**: Use with pre-configured dependencies ([Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/)) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are passing. This ensures verified functionality of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu.

This response is automated 🤖, but rest assured, an Ultralytics engineer will review and provide further assistance soon. Thank you for being a part of our community 😊!

**Comment by sato-kenryo** [comment]
Hello, thanks for the great Ultralytics documents and helpful advice.

And I encounterd additional issue when I tried to export "yolo11n-pose.pt" into "imx" format.
I reffered the [Ultralytics site](https://docs.ultralytics.com/ja/tasks/pose/#export) and run the codes on Google Colab.
```
from ultralytics import YOLO
model = YOLO("yolo11n-pose.pt")  # load an official model
model.export(format="imx")
```

But I got a TypeError about Tensor.
![WS000001](https://github.com/user-attachments/assets/53f4270c-55ef-4b23-ad21-e8a8fb761cd1)

Could anyone know the solution?

Thank you for your cooperation,

**Reply by Y-T-G** [reply]
Export is supported currently only for `yolov8n.pt`.
